engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2025,15 +1914,13 @@ - (NSArray*)selectionRectsForRange:(UITextRange*)range {
       }
       CGRect rect = CGRectMake(_selectionRects[i].rect.origin.x, _selectionRects[i].rect.origin.y,
                                width, _selectionRects[i].rect.size.height);
-      FlutterTextSelectionRect* selectionRect = [FlutterTextSelectionRect
-          selectionRectWithRectAndInfo:rect
-                              position:_selectionRects[i].position
-                      writingDirection:NSWritingDirectionNatural
-                         containsStart:(i == 0)
-                           containsEnd:(i == fml::RangeForCharactersInRange(
```

Comment:
curiuos why we needed to call this function before?

Reply:
I still don't understand what this is trying to do, I only inlined the `fml::RangeForCharactersInRange` called and deduced the logic to this.